### PR TITLE
Batch access list review reminders and provide link (slack)

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport.e/issues/3571

Instead of sending a reminder PER access list review due, it now only sends one (batched or singular) and includes link to the access list page. This required to gather all access list upfront (instead of processing reminders per page), putting it into a lookup map by recipients. 

If a recipient has >1 access list reviews due, the messaging is batched, with the earliest due date mentioned, and links out to the access list listing page:

image

If a recipient has only one review due, it keeps the previous messaging, including a link out to the specific access list:

image

changelog: For slack integration, Access List reminders are batched into 1 message and provides link out to the web UI.

### CODE REVIEW

File: integrations/access/accesslist/app.go [hunk 1]
```diff
@@ -212,10 +245,15 @@ func (a *App) notifyForAccessListReviews(ctx context.Context, accessList *access
...
+		return nil, trace.Wrap(err, "during create")
+	}
+
+	recipients, err := a.updatePluginDataAndGetRecipientsRequiringReminders(ctx, accessList, allRecipients, now, notificationStart)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
 
-	return trace.Wrap(a.sendMessages(ctx, accessList, allRecipients, now, notificationStart))
+	return recipients, err
```
Comment (smallinsky) [inline integrations/access/accesslist/app.go:256]:
```suggestion
	return recipients, nil
```

File: integrations/access/accesslist/app.go [hunk 2]
```diff
@@ -150,6 +150,8 @@ func (a *App) remindIfNecessary(ctx context.Context) error {
 
 	var nextToken string
 	var err error
+	remindersLookup := make(map[string][]*accesslist.AccessList)
+	recipientLookup := make(map[string]common.Recipient)
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply integrations/access/accesslist/app.go]: that's wild... made the change 👍

File: integrations/access/accesslist/app.go [hunk 3]
```diff
@@ -192,12 +225,12 @@ func (a *App) notifyForAccessListReviews(ctx context.Context, accessList *access
 	// If the current time before the notification start time, skip notifications.
 	if now.Before(notificationStart) {
 		log.Debugf("Access list %s is not ready for notifications, notifications start at %s", accessList.GetName(), notificationStart.Format(time.RFC3339))
-		return nil
+		return nil, nil
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply integrations/access/accesslist/app.go:219]: if i understood you correctly, iterating over a nil slice and spreading a nil slice, doesn't result in panic: https://go.dev/play/p/VbdQF2X6s5f

**Review by smallinsky — APPROVED** [review]

File: integrations/access/accesslist/app.go [hunk 4]
```diff
@@ -177,12 +189,25 @@ func (a *App) remindIfNecessary(ctx context.Context) error {
...
 
+	// Send reminders for each collected recipients.
+	var errs []error
+	for recipient, accessLists := range remindersLookup {
+		if err := a.bot.SendReviewReminders(ctx, recipient, accessLists); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply integrations/access/accesslist/app.go:200]: agh thanks!!!

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**